### PR TITLE
Rewrite ChainerX `batch_norm` test

### PR DESCRIPTION
**Pull request by hvy**
Partially fixes https://github.com/chainer/chainer/issues/6401

Also fixes a bug in `BatchNorm` for "None" upstream gradients.

### CODE REVIEW

File: tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import chainer
 import numpy
 import pytest
 
+import chainer
```
Comment (niboshi) [inline tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: It seems it should be at the original place?

Comment (hvy) [reply tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
It seems some relaxation of tolerances is needed:
```
_ test_BatchNorm_param_0_{contiguous=None}_param_0_{decay=None}_param_0_{eps=2e-05}_param_1_{x_shape=(2, 3, 4, 5), reduced_shape=(3, 4, 5), axis=(0,)}_backward[float64-native:0] _
```

```
E           gradients (numeric):  array(0.0233176, shape=(), dtype=float64, device='native:0')
E           gradients (backward): array(0.02311236, shape=(), dtype=float64, device='native:0')
E           
E           
E           Not equal to tolerance rtol=0.001, atol=0.0001
E           
E           Mismatch: 100%
E           Max absolute difference: 0.00020525
E           Max relative difference: 0.00888048
E            x: array(0.023318)
E            y: array(0.023112)
E           
E           assert_allclose failed: 
E             shape: () ()
E             dtype: float64 float64
E             i: (0,)
E             x[i]: 0.02331760894588602
E             y[i]: 0.02311236000087985
E             relative error[i]: 0.008880484078577661
E             absolute error[i]: 0.0002052489450061687
E           x: 0.02331761
E           y: 0.02311236
```

File: tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py [hunk 2]
```diff
@@ -67,57 +68,137 @@ def _create_batch_norm_ndarray_args(
...
+        # - Non-contiguous running values which are updated in-place are not
+        # supported by CUDA.
+        # - Non-contiguous gamma and beta is not supported by CUDA.
+        # TODO(hvy): Support non-contiguous gamma and beta with CUDA. Create a
+        # contiguous copy in the cuDNN wrapper.
+        self.is_cuda = chainerx.get_default_device().backend.name == 'cuda'
+        if self.is_cuda and self.contiguous is None:
+            self.skip_forward_test = True
+            self.skip_backward_test = True
+            self.skip_double_backward_test = True
```
Comment (niboshi) [inline tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: It's better to `raise unittest.SkipTest('...')`.

Comment (hvy) [reply tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: Fixed.

**Comment by hvy** [comment]
Tried halving the tolerances for backward. PTAL.

File: tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def setup(self, float_dtype):
         else:
             self.check_forward_options.update({'rtol': 1e-6, 'atol': 1e-5})
             self.check_backward_options.update({
-                'eps': 1e-3, 'rtol': 1e-3, 'atol': 1e-4})
+                'eps': 1e-3, 'rtol': 5e-2, 'atol': 5e-3})
```
Comment (niboshi) [inline tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: They're 50 times larger. I guess that's not what's intended.

Comment (hvy) [reply tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: Sorry, let me fix it.

Comment (hvy) [reply tests/chainerx_tests/unit_tests/routines_tests/test_normalization.py]: Fixed. Made if 5 times larger.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM
